src/Nav/cprt_gridmap_filters/src/PreserveCostInflationFilter.cpp

```diff
@@ -80,10 +81,38 @@ template <typename T> bool PreserveCostInflationFilter<T>::configure() {
         "PreserveCostInflationFilter parameter 'decay_rate' must be >= 0.0.");
     return false;
   }
-
   return true;
 }
 
+template <typename T>
+void PreserveCostInflationFilter<T>::create_lookup_table(const T &mapOut) {
+
+  int maxDx_ = static_cast<int>(std::ceil(decayInflationRadius_) /
+                                mapOut.getResolution());
+  if (maxDx_ + 1 == decayTable_.size()) {
+    return;
+  }
+  // prepare table (square)
+  decayTable_.resize(maxDx_ + 1, std::vector<float>(maxDx_ + 1, 0.0));
+
+  // fill up tables with values
+  for (int dx = 0; dx <= maxDx_; ++dx) {
+    for (int dy = 0; dy <= maxDx_; ++dy) {
+      float distance = std::sqrt(dx * dx + dy * dy) * mapOut.getResolution();
+      if (distance <=
+          coreInflationRadius_) { // inside inflation radius so no decay
```

Comment:
```suggestion
          coreInflationRadius_) {
```

Comment self explanitory from the code